src/argilla/client/feedback/dataset/remote.py

```diff
@@ -0,0 +1,257 @@
+
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional, Union
+
+from tqdm import trange
+
+from argilla.client.feedback.constants import FETCHING_BATCH_SIZE, PUSHING_BATCH_SIZE
+from argilla.client.feedback.dataset.base import FeedbackDatasetBase
+from argilla.client.feedback.schemas.records import FeedbackRecord, RemoteFeedbackRecord
+from argilla.client.sdk.v1.datasets import api as datasets_api_v1
+
+if TYPE_CHECKING:
+    from uuid import UUID
+
+    import httpx
+
+    from argilla.client.feedback.dataset.local import FeedbackDataset
+    from argilla.client.feedback.types import AllowedFieldTypes, AllowedQuestionTypes
+    from argilla.client.sdk.v1.datasets.models import FeedbackItemModel
+    from argilla.client.workspaces import Workspace
+
+
+warnings.simplefilter("always", DeprecationWarning)
+
+
+class RemoteFeedbackRecords:
+    def __init__(self, client: "httpx.Client", id: "UUID", questions: List["AllowedQuestionTypes"]) -> None:
+        self.client = client
+        self.id = id
+
+        self.__question_id2name = {question.id: question.name for question in questions}
+        self.__question_name2id = {value: key for key, value in self.__question_id2name.items()}
+
+    def __repr__(self) -> str:
+        return (
+            "The `records` of a `FeedbackDataset` in Argilla are being lazily"
+            " fetched, and never stored locally. You can either loop over `records`"
+            " or access them by index, and those will be fetched from Argilla on the"
+            " fly."
+        )
+
+    def __parse_record(self, record: "FeedbackItemModel") -> RemoteFeedbackRecord:
+        record = record.dict(
+            exclude={
+                "inserted_at": ...,
+                "updated_at": ...,
+                "responses": {"__all__": {"id", "inserted_at", "updated_at"}},
+                "suggestions": {"__all__": {"id"}},
+            },
+            exclude_none=True,
+        )
+        for suggestion in record.get("suggestions", []):
+            suggestion.update({"question_name": self.__question_id2name[suggestion["question_id"]]})
+        return RemoteFeedbackRecord(client=self.client, name2id=self.__question_name2id, **record)
+
+    def __len__(self) -> int:
+        try:
+            response = datasets_api_v1.get_metrics(client=self.client, id=self.id)
+        except Exception as e:
+            raise Exception(
+                f"Failed while getting the metrics from the current `FeedbackDataset` in Argilla with exception: {e}"
+            ) from e
+        return response.parsed.records.count
+
+    def __getitem__(self, key: Union[slice, int]) -> Union[RemoteFeedbackRecord, List[RemoteFeedbackRecord]]:
+        offsets = []
+        limit = None
+        if isinstance(key, slice):
+            start, stop, step = key.indices(len(self))
+            if step is not None and step != 1:
+                return [self[i] for i in range(start, stop, step)]
```

Comment:
This case will call several times the API to fetch records one-by-one right? I would say start by not supporting slices with `step` != 1 for remote datasets.

Reply:
But what if that's what the user wants? At the end we want to expose the records so that the user can use it as a normal list but inside it's fetching from Argilla, so I think we can leave it as is, at the end it depends on the user to avoid doing that